tests/Aspire.Dashboard.Tests/Integration/Playwright/PlaywrightFixture.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Workload.Tests;
+using Microsoft.Playwright;
+using Polly;
+using Polly.Retry;
+using Xunit;
+
+namespace Aspire.Dashboard.Tests.Integration.Playwright;
+
+public class PlaywrightFixture : IAsyncLifetime
+{
+    public IBrowser Browser { get; set; } = null!;
+    private ResiliencePipeline _resiliencePipeline = null!;
+
+    public async Task InitializeAsync()
+    {
+
+        var retryOptions = new RetryStrategyOptions
+        {
+            ShouldHandle = new PredicateBuilder().Handle<ArgumentException>(),
+            BackoffType = DelayBackoffType.Exponential,
+            UseJitter = true,
+            MaxRetryAttempts = 20,
+            Delay = TimeSpan.FromMilliseconds(100),
+        };
+
+        var resiliencePipelineBuilder = new ResiliencePipelineBuilder();
+        resiliencePipelineBuilder
+            .AddRetry(retryOptions)
+            .AddTimeout(TimeSpan.FromSeconds(10));
+        _resiliencePipeline = resiliencePipelineBuilder.Build();
+
+        Browser = await PlaywrightProvider.CreateBrowserAsync();
+    }
+
+    public async Task DisposeAsync()
+    {
+        await Browser.CloseAsync();
+    }
+
+    public async Task GoToHomeAndWaitForDataGridLoad(IPage page)
+    {
+        await page.GotoAsync("/");
+
+        await _resiliencePipeline.ExecuteAsync(async _ =>
```

Comment:
Is this like a custom retry strategy? I recommend to e.g. stick with ToBeVisible which will retry and end up in better reporting in e.g. Trace Viewer or logs.

https://playwright.dev/dotnet/docs/api/class-locatorassertions#locator-assertions-to-be-visible

Reply:
Thanks, I've made this change 😄